libsolidity/formal/CHC.cpp

```diff
@@ -707,6 +725,59 @@ void CHC::visitAddMulMod(FunctionCall const& _funCall)
 	SMTEncoder::visitAddMulMod(_funCall);
 }
 
+void CHC::visitDeployment(FunctionCall const& _funCall)
+{
+	if (!externalCallsIsTrustedMode())
+	{
+		SMTEncoder::endVisit(_funCall);
```

Comment:
This doesn't additionally need ``unknownFunctionCall(_funCall)`` as before?

Reply:
Yep, state should be erased and we should have tests that use `new` and assert things about state variables